### PR TITLE
feat: mount asgi app and fastapi support

### PR DESCRIPTION
**Pull request by frostming**
Signed-off-by: Frost Ming <me@frostming.com>

## What does this PR address?

Fixes #(issue)

## Before submitting:

- [ ] Does the Pull Request follow [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary) naming? Here are [GitHub's
      guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on how to create a pull request.
- [ ] Does the code follow BentoML's code style, `pre-commit run -a` script has passed ([instructions](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#style-check-auto-formatting-type-checking))?
- [ ] Did you read through [contribution guidelines](https://github.com/bentoml/BentoML/blob/main/CONTRIBUTING.md#ways-to-contribute) and follow [development guidelines](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#start-developing)?
- [ ] Did your changes require updates to the documentation? Have you updated
      those accordingly? Here are [documentation guidelines](https://github.com/bentoml/BentoML/tree/main/docs) and [tips on writting docs](https://github.com/bentoml/BentoML/tree/main/docs#writing-documentation).
- [ ] Did you write tests to cover your changes?

### CODE REVIEW

**Review by bojiang — APPROVED** [review]

**Review by bojiang — CHANGES_REQUESTED** [review]
dynamically import fastapi

File: pyproject.toml [hunk 1]
```diff
@@ -166,6 +166,7 @@ testing = [
     "scikit-learn",
     "yamllint==1.32.0",
     "coverage[toml]==7.2.6",
+    "fastapi",
```
Comment (bojiang) [inline pyproject.toml:169]: We can use https://github.com/bentoml/BentoML/blob/main/src/bentoml/_internal/types.py#L83 to avoid hard depends on fastapi

Comment (frostming) [reply pyproject.toml:169]: that's a testing only dependency. to test fastapi mount behavior

Comment (aarnphm) [reply pyproject.toml:169]: I think test is fine for this.

**Pull request merged**